crates/nu-command/src/math/utils.rs

```diff
@@ -94,3 +94,86 @@ pub fn calculate(
         )),
     }
 }
+
+// Utility functions to reduce boiler plate in the the
+// trigonometric math commands
+pub fn trig_value<F>(value: Value, use_degrees: Option<bool>, f: F) -> Value
+where
+    F: Fn(f64) -> f64,
+{
+    let (val, span) = match value {
+        Value::Int { val, span } => (val as f64, span),
+        Value::Float { val, span } => (val, span),
+        _ => unreachable!(),
```

Comment:
Using `unreachable!()` in the context of a free function is wrong in this context by convention.

It will still panic if used but is not expressing a human-proven condition that can help as a compiler/user hint as in the double match. Instead it should clearly communicate its constraints.

Maybe this is a clear signal that the boundary for splitting into functions should be a bit different to enable a more useful signature.